### ISSUE TITLE
[Godot 4 alpha 5] Editor crashed when setting font size to bigger

### ISSUE THREAD

**Issue by njdf18** [godotengine/godot#59794]
### Godot version

v4.0.alpha5.official [d7d528c15]

### System information

Windows11, Vulkan, NVIDIA GT 1050ti

### Issue description

Godot Editor crashed when setting font size to bigger.

### Steps to reproduce

1. Create an empty project and scene, and add "RichTextLabel" node.
2. Create a "Font" resource on the node, with any OTF or TTF.
3. Change the value of font size to bigger.

![godotBug1](https://user-images.githubusercontent.com/86307226/161282732-ae2f00ed-3e26-404b-b21c-c23b57dcc172.gif)

**Comment by qarmin** [comment]
Can't reproduce with fd0716cba
Please provide minimal project

**Comment by njdf18** [comment]
OK, But I don't know why I can't attach .zip to Github(I got "is not included in the list"), so I placed it on Google Drive.
https://drive.google.com/file/d/1dN2PP9vmpd_9KYCPtDFjLwVM96uo8azR/view?usp=sharing

**Comment by njdf18** [comment]
I also found out the font that only support English character won't make the editor crash when I changing the size.

**Comment by njdf18** [comment]
I tested this project on Linux Mint, but it won't crash when doing the same behaviour.

**Comment by bruvzg** [comment]
Can't reproduce, font size is limited by texture's size (1024 x 1024), but even it is won't fit its printing errors, but won't crash.

**Comment by dardanbujupaj** [comment]
I got the same Issue with Godot v3.5.beta4. It seems to be also present in beta2 and beta3 but not in beta1.

I cannot reproduce it on Linux and MacOS so it seems to be exclusive to Windows.

Video ends, when Godot crashes:
https://user-images.githubusercontent.com/16074028/163862544-79fe17b0-3222-4ce7-8a58-9c70586952e9.mp4

Minimal reproduction:
[large-font-minimal-reproduction.zip](https://github.com/godotengine/godot/files/8507648/large-font-minimal-reproduction.zip)

**Comment by dardanbujupaj** [comment]
Maybe it has something to do with the freetype update? 
https://github.com/godotengine/godot/pull/55124

This was also cherrypicked for 3.5.beta2.

**Comment by dardanbujupaj** [comment]
I just built Godot from source (on Windows with MinGW) and wasn't able to reproduce it.

I tried:
- master
- 3.x
- https://github.com/godotengine/godot/commit/b6968ab0602bfe72c71d4efcafe608f9cac36252 (wich is the commit 3.5.beta4 is built with)

I also tried with target=debug_release. 

None of them crashed with methods above 🤔.

I wonder if this has something to do with the way the prerelease binaries are built?

**Comment by dardanbujupaj** [comment]
Nevermind... it does crash with `use_lto=yes`

**Comment by dardanbujupaj** [comment]
> Maybe it has something to do with the freetype update? 
> 
> https://github.com/godotengine/godot/pull/55124
> 
> 
> 
> This was also cherrypicked for 3.5.beta2.

I'm pretty sure it's the freetype version at this point.
When I reverted the commit from the PR on the 3.x branch it didn't crash anymore. 
I also tried to update to freetype to 2.12.0 but hat some conflicts in the linking phase. 

Couldn't test on master though because I didn't manage to get a running build with the `use_lto` option active. 

I'll give it another try tomorrow but if anybody has inputs regarding the two building issues I faced I'd be very thankful 😁
